org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java

```diff
@@ -6472,7 +6483,8 @@ public static String latestSupportedJavaVersion() {
 	 * @since 3.39
 	 */
 	public static String getFirstJavaSourceVersionSupportedByCompiler() {
-		return SUPPORTED_VERSIONS.get(0);
+		return SUPPORTED_VERSIONS.first();
+		//TODO: drop this again and just mention the use-case in the doc of getAllJavaSourceVersionsSupportedByCompiler()?
```

Comment:
This could be done in a second PR, because currently JDT UI uses this method. Once this PR is merged, we could use `getAllJavaVersionsSupportedByCompiler().first()`.